### ISSUE TITLE
Contrary to docs, cannot supply plain string as message

### ISSUE THREAD

**Issue by HannesP** [Blizzard/node-rdkafka#199]
According to the docs, a plain string can be fed to Producer#produce for the message body, and will be converted to a Buffer object automatically:

```js
producer.produce(
  // Topic to send the message to
  'topic',
  // optionally we can manually specify a partition for the message
  // this defaults to -1 - which will use librdkafka's default partitioner (consistent random for keyed messages, random for unkeyed messages)
  null,
  // Message to send. If a string is supplied, it will be
  // converted to a Buffer automatically, but we're being
  // explicit here for the sake of example.
  new Buffer('Awesome message'),
  // for keyed messages, we also specify the key - note that this field is optional
  'Stormwind',
  // you can send a timestamp here. If your broker version supports it,
  // it will get added. Otherwise, we default to 0
  Date.now(),
  // you can send an opaque token here, which gets passed along
  // to your delivery reports
);
```

However, when I do this, I get an `Error: Message must be a buffer or null` exception. When feeding a Buffer object it works as expected. Are the docs outdated or am I missing something?

**Comment by webmakersteve** [comment]
I'll update the docs. You should always pass a buffer in for the message because I would need to convert it to bytes in C++ land anyway.

**Comment by webmakersteve** [comment]
Fixed with b2e4d1c56145d57d3bce05c2dcc42565bbc4f84e
